### PR TITLE
Generates docs for ndarrays

### LINKED ISSUE #7531: [hail][ndarray] NDArray Documentation
NDArrays don't yet generate documentation on hail website

### PR DESCRIPTION
**Pull request by johnc1231**
This PR generates documentation for all current ndarray functionality. Wondering if to coincide with this we should also remove the `_` from `hl._nd`. I mostly had it there so people wouldn't poke at undocumented functionality, but this seems to work ok now. 

Resolves #7531

### CODE REVIEW

File: hail/python/hail/expr/expressions/typed_expressions.py [hunk 1]
```diff
@@ -3782,75 +3783,75 @@ def __matmul__(self, other):
...
-        Returns
-        -------
-        :class:`numpy.ndarray`
-        """
-        # FIXME Use filesystem abstraction instead when that is ready
-        temp_file = tempfile.NamedTemporaryFile(suffix='.npy').name
-        self.save(temp_file)
-        return np.load(temp_file)
+    # @typecheck_method(axis=nullable(oneof(int, sequenceof(int))))
+    # def sum(self, axis=None):
```
Comment (patrick-schultz) [inline hail/python/hail/expr/expressions/typed_expressions.py]: ?

File: hail/python/hail/nd/nd.py [hunk 2]
```diff
@@ -67,26 +124,30 @@ def zeros(shape, dtype=hl.tfloat64):
...
-       It is possible to specify a type other than `tfloat64` with the `dtype` argument.
+       It is possible to specify a type other than :py:data:`.tfloat64` with the `dtype` argument.
 
        >>> hl._nd.zeros((5, 7), dtype=hl.tfloat32)
 
 
        Parameters
        ----------
-       shape : `tuple` or `TupleExpression`
+       shape: `tuple` or :class:`.TupleExpression`
```
Comment (patrick-schultz) [inline hail/python/hail/nd/nd.py]: space before :

File: hail/python/hail/expr/expressions/typed_expressions.py (hunk 1 above)
Comment (johnc1231) [reply hail/python/hail/expr/expressions/typed_expressions.py]: Not currently implemented, so I don't want it to show up on the `NDArrayExpression` class.

Comment (johnc1231) [reply hail/python/hail/expr/expressions/typed_expressions.py]: I have not implemented `NDArrayAgg` IR Node

Comment (patrick-schultz) [reply hail/python/hail/expr/expressions/typed_expressions.py]: Ah. Having a big block of commented out code is confusing. Could you save this in a branch and add it back when `NDArrayAgg` is implemented? Either that, or uncomment it, and have throw a "not implemented" error.

**Pull request merged**